core/src/test/java/hudson/FunctionsTest.java

```diff
@@ -369,6 +371,34 @@ private void assertBrokenAs(String plain, String... chunks) {
         assertEquals("&lt;discarded/&gt;\n", Functions.printLogRecordHtml(lr, null)[3]);
     }
 
+    @Test
+    public void extractPluginNameFromIconSrcHandlesNull() {
+        String result = Functions.extractPluginNameFromIconSrc(null);
+
+        assertThat(result, is(emptyString()));
+    }
+
+    @Test
+    public void extractPluginNameFromIconSrcHandlesEmptyString() {
+        String result = Functions.extractPluginNameFromIconSrc(null);
+
+        assertThat(result, is(emptyString()));
+    }
+
+    @Test
+    public void extractPluginNameFromIconSrcOnlyReturnsPluginFromStart() {
```

Comment:
I must admit that from the code and test I do not actually understand how the input parameter should look like? Are these always two space separated String values? The first one the icon, the second one the plugin? Maybe it would be helpful if this is documented somewhere. Or is this explained in the design library?

Reply:
It is in the design library